// romcal/dist/calendars/general.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.dates = void 0;

var _moment = _interopRequireDefault(require("moment"));

var _lodash = _interopRequireDefault(require("lodash"));

var _lib = require("../lib");

var _constants = require("../constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var dates = function dates(year) {
  var _dates = [{
    "key": "saintsBasilTheGreatAndGregoryNazianzenBishopsAndDoctors",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 2
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "theMostHolyNameOfJesus",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 3
    }),
    "data": {}
  }, {
    "key": "saintRaymondOfPenyafortPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 7
    }),
    "data": {}
  }, {
    "key": "saintHilaryOfPoitiersBishopAndDoctor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 13
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintAnthonyOfEgyptAbbot",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 17
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintsFabianPopeAndSebastianMartyrs",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 20
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintAgnesVirginAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 21
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintVincentDeaconAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 22
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintFrancisDeSalesBishopAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 24
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, // The proper color for the Chair of Peter (Feast, Feb 22) and the Conversion of
  //cSt. Paul (Feast, Jan 25) is white, although both St. Peter and St. Paul
  // were martyrs.
  {
    "key": "conversionOfSaintPaulApostle",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 25
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintsTimothyAndTitusBishops",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 26
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintAngelaMericiVirgin",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 27
    }),
    "data": {}
  }, {
    "key": "saintThomasAquinasPriestAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 28
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintJohnBoscoPriest",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 0,
      day: 31
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintBlaseBishopAndMartyrSaintAnsgarBishop",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 3
    }),
    "data": {}
  }, {
    "key": "saintAgathaVirginAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 5
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintPaulMikiAndCompanionsMartyrs",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 6
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintJeromeEmilianiSaintJosephineBakhitaVirgin",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 8
    }),
    "data": {}
  }, {
    "key": "saintScholasticaVirgin",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 10
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "ourLadyOfLourdes",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 11
    }),
    "data": {}
  }, {
    "key": "saintsCyrilMonkAndMethodiusBishop",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 14
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.PATRON_OF_EUROPE]
      }
    }
  }, {
    "key": "sevenHolyFoundersOfTheServiteOrder",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 17
    }),
    "data": {}
  }, {
    "key": "saintPeterDamianBishopAndDoctorOfTheChurch",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 21
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "chairOfSaintPeterApostle",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 22
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintPolycarpBishopAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 1,
      day: 23
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintCasimir",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 2,
      day: 4
    }),
    "data": {}
  }, {
    "key": "saintsPerpetuaAndFelicityMartyrs",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 2,
      day: 7
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintJohnOfGodReligious",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 2,
      day: 8
    }),
    "data": {}
  }, {
    "key": "saintFrancesOfRomeReligious",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 2,
      day: 9
    }),
    "data": {}
  }, {
    "key": "saintPatrickBishop",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 2,
      day: 17
    }),
    "data": {}
  }, {
    "key": "saintCyrilOfJerusalemBishopAndDoctor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 2,
      day: 18
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintTuribiusOfMogrovejoBishop",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 2,
      day: 23
    }),
    "data": {}
  }, {
    "key": "saintFrancisOfPaolaHermit",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 2
    }),
    "data": {}
  }, {
    "key": "saintIsidoreOfSevilleBishopAndDoctorOfTheChurch",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 4
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintVincentFerrerPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 5
    }),
    "data": {}
  }, {
    "key": "saintJohnBaptistDeLaSallePriest",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 7
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintStanislausBishopAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 11
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintMartinIPopeAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 13
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintAnselmOfCanterburyBishopAndDoctorOfTheChurch",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 21
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintGeorgeMartyrSaintAdalbertBishopAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 23
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintFidelisOfSigmaringenPriestAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 24
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintMarkTheEvangelist",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 25
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintPeterChanelPriestAndMartyrSaintLouisGrignonDeMontfortPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 28
    }),
    "data": {}
  }, {
    "key": "saintCatherineOfSienaVirginAndDoctorOfTheChurch",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 29
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.PATRON_OF_EUROPE, _constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintPiusVPope",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 3,
      day: 30
    }),
    "data": {}
  }, {
    "key": "saintJosephTheWorker",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 1
    }),
    "data": {}
  }, {
    "key": "saintAthanasiusBishopAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 2
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintsPhilipAndJamesApostles",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 3
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintsNereusAndAchilleusMartyrsSaintPancrasMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 12
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "ourLadyOfFatima",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 13
    }),
    "data": {}
  }, {
    "key": "saintMatthiasTheApostle",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 14
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintJohnIPopeAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 18
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintBernardineOfSienaPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 20
    }),
    "data": {}
  }, {
    "key": "saintChristopherMagallanesAndCompanionsMartyrs",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 21
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintRitaOfCascia",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 22
    }),
    "data": {}
  }, {
    "key": "saintBedeTheVenerablePriestAndDoctorSaintGregoryViiPopeSaintMaryMagdaleneDePazziVirgin",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 25
    }),
    "data": {}
  }, {
    "key": "saintPhilipNeriPriest",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 26
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintAugustineOfCanterburyBishop",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 27
    }),
    "data": {}
  }, {
    "key": "visitationOfTheBlessedVirginMary",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 4,
      day: 31
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintJustinMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 1
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintsMarcellinusAndPeterMartyrs",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 2
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintsCharlesLwangaAndCompanionsMartyrs",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 3
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintBonifaceBishopAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 5
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintNorbertBishop",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 6
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintEphremDeaconAndDoctor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 9
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintBarnabasTheApostle",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 11
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintAnthonyOfPaduaPriestAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 13
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintRomualdAbbot",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 19
    }),
    "data": {}
  }, {
    "key": "saintAloysiusGonzagaReligious",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 21
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintPaulinusOfNolaBishopSaintsJohnFisherBishopAndThomasMoreMartyrs",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 22
    }),
    "data": {}
  }, {
    "key": "saintCyrilOfAlexandriaBishopAndDoctor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 27
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintIrenaeusBishopAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 28
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "firstMartyrsOfTheChurchOfRome",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 5,
      day: 30
    }),
    "data": {}
  }, {
    "key": "saintThomasTheApostle",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 3
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintElizabethOfPortugal",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 4
    }),
    "data": {}
  }, {
    "key": "saintAnthonyZaccariaPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 5
    }),
    "data": {}
  }, {
    "key": "saintMariaGorettiVirginAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 6
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintsAugustineZhaoRongPriestAndCompanionsMartyrs",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 9
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintBenedictOfNursiaAbbot",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 11
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.PATRON_OF_EUROPE]
      }
    }
  }, {
    "key": "saintHenryBishopAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 13
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintCamillusDeLellisPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 14
    }),
    "data": {}
  }, {
    "key": "saintBonaventureBishopAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 15
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "ourLadyOfMountCarmel",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 16
    }),
    "data": {}
  }, {
    "key": "saintApollinaris",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 20
    }),
    "data": {}
  }, {
    "key": "saintLawrenceOfBrindisiPriestAndDoctor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 21
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, // https://github.com/pejulian/romcal/issues/27
  {
    "key": "saintMaryMagdalene",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 22
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, // On 11 Feb 2018, Pope Francis inserted the Memorial of the
  // Blessed Virgin Mary, Mother of the Church, into the General Roman Calendar.
  // It will be celebrated on Monday after Pentecost.
  {
    "key": "maryMotherOfTheChurch",
    "type": _constants.Types.MEMORIAL,
    // Memorial
    "moment": function (y) {
      return _lib.Dates.pentecostSunday(y).add(1, 'days');
    }(year),
    "data": {
      "prioritized": true,
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintBridgetOfSwedenReligious",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 23
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.PATRON_OF_EUROPE]
      }
    }
  }, {
    "key": "saintCharbelMakhloufPriestAndHermit",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 24
    }),
    "data": {}
  }, {
    "key": "saintJamesApostle",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 25
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintsJoachimAndAnne",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 26
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintMartha",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 29
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintPeterChrysologusBishopAndDoctor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 30
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintIgnatiusOfLoyolaPriest",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 6,
      day: 31
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintAlphonsusMariaDeLiguoriBishopAndDoctorOfTheChurch",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 1
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintEusebiusOfVercelliBishopSaintPeterJulianEymardPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 2
    }),
    "data": {}
  }, {
    "key": "saintJeanVianneyTheCureOfArsPriest",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 4
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "dedicationOfTheBasilicaOfSaintMaryMajor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 5
    }),
    "data": {}
  }, {
    "key": "saintSixtusIiPopeAndCompanionsMartyrsSaintCajetanPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 7
    }),
    "data": {}
  }, {
    "key": "saintDominicPriest",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 8
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintTeresaBenedictaOfTheCrossEdithSteinVirginAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 9
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR, _constants.Titles.PATRON_OF_EUROPE]
      }
    }
  }, {
    "key": "saintLawrenceOfRomeDeaconAndMartyr",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 10
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintClareVirgin",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 11
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintJaneFrancesDeChantalReligious",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 12
    }),
    "data": {}
  }, {
    "key": "saintsPontianPopeAndHippolytusPriestMartyrs",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 13
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintMaximilianMaryKolbePriestAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 14
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintStephenOfHungary",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 16
    }),
    "data": {}
  }, {
    "key": "saintJohnEudesPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 19
    }),
    "data": {}
  }, {
    "key": "saintBernardOfClairvauxAbbotAndDoctorOfTheChurch",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 20
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintPiusXPope",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 21
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "queenshipOfBlessedVirginMary",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 22
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintRoseOfLima",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 23
    }),
    "data": {}
  }, {
    "key": "saintBartholomewTheApostle",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 24
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintLouisSaintJosephOfCalasanzPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 25
    }),
    "data": {}
  }, {
    "key": "saintMonica",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 27
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintAugustineOfHippoBishopAndDoctorOfTheChurch",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 28
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "theBeheadingOfSaintJohnTheBaptistMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 7,
      day: 29
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintGregoryTheGreatPopeAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 3
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "birthOfTheBlessedVirginMary",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 8
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintPeterClaverPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 9
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "holyNameOfTheBlessedVirginMary",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 12
    }),
    "data": {}
  }, {
    "key": "saintJohnChrysostomBishopAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 13
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "theExaltationOfTheHolyCross",
    "type": _constants.Types[4],
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 14
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED
      }
    }
  }, {
    "key": "ourLadyOfSorrows",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 15
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintsCorneliusPopeAndCyprianBishopMartyrs",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 16
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintRobertBellarmineBishopAndDoctor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 17
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintJanuariusBishopAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 19
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintAndrewKimTaegonPriestAndPaulChongHasangAndCompanionsMartyrs",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 20
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintMatthewApostleAndEvangelist",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 21
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintPioOfPietrelcinaPadrePioPriest",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 23
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintsCosmasAndDamianMartyrs",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 26
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintVincentDePaulPriest",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 27
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintWenceslausMartyrSaintsLawrenceRuizAndCompanionsMartyrs",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 28
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintsMichaelGabrielAndRaphaelArchangels",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 29
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintJeromePriestAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 8,
      day: 30
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintThereseOfTheChildJesusVirginAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 1
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "guardianAngels",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 2
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintFrancisOfAssisi",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 4
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintBrunoPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 6
    }),
    "data": {}
  }, {
    "key": "ourLadyOfTheRosary",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 7
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintDenisAndCompanionsMartyrsSaintJohnLeonardiPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 9
    }),
    "data": {}
  }, // http://www.vatican.va/roman_curia/congregations/ccdds/documents/rc_con_ccdds_doc_20140529_decreto-calendario-generale-gxxiii-gpii_en.html
  {
    "key": "popeSaintJohnXXIII",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 11
    }),
    "data": {}
  }, // http://www.vatican.va/roman_curia/congregations/ccdds/documents/rc_con_ccdds_doc_20140529_decreto-calendario-generale-gxxiii-gpii_en.html
  {
    "key": "popeSaintJohnPaulII",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 22
    }),
    "data": {}
  }, {
    "key": "saintCallistusIPopeAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 14
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintTeresaOfJesusVirginAndDoctorOfTheChurch",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 15
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintHedwigReligiousSaintMargaretMaryAlacoqueVirgin",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 16
    }),
    "data": {}
  }, {
    "key": "saintIgnatiusOfAntiochBishopAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 17
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintLukeTheEvangelist",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 18
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintsJeanDeBrebeufAndIsaacJoguesPriestsAndCompanionsMartyrsSaintPaulOfTheCrossPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 19
    }),
    "data": {}
  }, {
    "key": "saintJohnOfCapistranoPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 23
    }),
    "data": {}
  }, {
    "key": "saintAnthonyMaryClaretBishop",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 24
    }),
    "data": {}
  }, {
    "key": "saintsSimonAndJudeApostles",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 9,
      day: 28
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "allSouls",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 2
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintMartinDePorresReligious",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 3
    }),
    "data": {}
  }, {
    "key": "saintCharlesBorromeoBishop",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 4
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, // Replaces 32nd Sunday in Ordinary Time when it falls on a Sunday
  {
    "key": "dedicationOfTheLateranBasilica",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 9
    }),
    "data": {
      "prioritized": true,
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintLeoTheGreatPopeAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 10
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintMartinOfToursBishop",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 11
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintJosaphatBishopAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 12
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintAlbertTheGreatBishopAndDoctor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 15
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintMargaretOfScotlandSaintGertrudeTheGreatVirgin",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 16
    }),
    "data": {}
  }, {
    "key": "saintElizabethOfHungary",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 17
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "dedicationOfTheBasilicasOfSaintsPeterAndPaulApostles",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 18
    }),
    "data": {}
  }, {
    "key": "presentationOfTheBlessedVirginMary",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 21
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintCeciliaVirginAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 22
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintClementIPopeAndMartyrSaintColumbanReligious",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 23
    }),
    "data": {}
  }, {
    "key": "saintAndrewDungLacAndCompanionsMartyrs",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 24
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintCatherineOfAlexandriaVirginAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 25
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintAndrewTheApostle",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 10,
      day: 30
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintFrancisXavierPriest",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 3
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "saintJohnDamascenePriestAndDoctor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 4
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintNicholasBishop",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 6
    }),
    "data": {}
  }, {
    "key": "saintAmbroseBishopAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 7
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintJuanDiegoCuauhtlatoatzin",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 9
    }),
    "data": {}
  }, {
    "key": "ourLadyOfLoreto",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 10
    }),
    "data": {}
  }, {
    "key": "saintDamasusIPope",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 11
    }),
    "data": {}
  }, {
    "key": "ourLadyOfGuadalupe",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 12
    }),
    "data": {}
  }, {
    "key": "saintLucyOfSyracuseVirginAndMartyr",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 13
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintJohnOfTheCrossPriestAndDoctor",
    "type": _constants.Types.MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 14
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE,
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintPeterCanisiusPriestAndDoctor",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 21
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.DOCTOR_OF_THE_CHURCH]
      }
    }
  }, {
    "key": "saintJohnOfKantyPriest",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 23
    }),
    "data": {}
  }, {
    "key": "saintStephenTheFirstMartyr",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 26
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintJohnTheApostleAndEvangelist",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 27
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.WHITE
      }
    }
  }, {
    "key": "holyInnocentsMartyrs",
    "type": _constants.Types.FEAST,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 28
    }),
    "data": {
      "meta": {
        "liturgicalColor": _constants.LiturgicalColors.RED,
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintThomasBecketBishopAndMartyr",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 29
    }),
    "data": {
      "meta": {
        "titles": [_constants.Titles.MARTYR]
      }
    }
  }, {
    "key": "saintSylvesterIPope",
    "type": _constants.Types.OPT_MEMORIAL,
    "moment": _moment["default"].utc({
      year: year,
      month: 11,
      day: 31
    }),
    "data": {}
  }];
  return _lodash["default"].map(_dates, function (_ref) {
    var key = _ref.key,
        data = _ref.data,
        rest = _objectWithoutProperties(_ref, ["key", "data"]);

    var name = _lib.Utils.localize({
      key: 'sanctoral.' + key
    });

    var meta = data.meta;

    if (_lodash["default"].isUndefined(meta)) {
      data.meta = {
        titles: []
      };
    }

    return _objectSpread({
      name: name,
      key: key,
      data: data
    }, rest);
  });
};

exports.dates = dates;
