// romcal/dist/lib/Seasons.js
"use strict";

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.laterOrdinaryTime = exports.eastertide = exports.easterOctave = exports.easterTriduum = exports.lent = exports.earlyOrdinaryTime = exports.christmastide = exports.advent = void 0;

var _moment = _interopRequireDefault(require("moment"));

var _lodash = _interopRequireDefault(require("lodash"));

var Dates = _interopRequireWildcard(require("./Dates"));

var Utils = _interopRequireWildcard(require("./Utils"));

var _constants = require("../constants");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

//================================================================================================
// METHODS TO GENERATE THE SEASONS
// PRIVATE: Not exposed via module exports
//================================================================================================
// dates: array of moment object dates
var _metadata = function _metadata(dates) {
  return _lodash["default"].map(dates, function (date) {
    date.source = 'l';
    date.data.calendar = {
      weeks: date.moment.weeksInYear(),
      week: date.moment.week(),
      day: date.moment.dayOfYear()
    };
    return date;
  });
}; // y: The year (integer)
// epiphanyOnJan6: true|false [If true, Epiphany will be fixed to Jan 6] (defaults to false)


var _epiphany = function _epiphany(y) {
  var epiphanyOnJan6 = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
  var before = Dates.daysBeforeEpiphany(y, epiphanyOnJan6);
  var after = Dates.daysAfterEpiphany(y, epiphanyOnJan6);
  var days = [];

  _lodash["default"].each(before, function (day) {
    days.push({
      moment: day,
      type: _constants.Types.FERIA,
      name: Utils.localize({
        key: 'epiphany.before',
        day: day.format('dddd')
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.CHRISTMASTIDE,
          value: Utils.localize({
            key: 'christmastide.season'
          })
        },
        meta: {
          liturgicalColor: '',
          titles: []
        }
      }
    });
  });

  _lodash["default"].each(after, function (day) {
    days.push({
      moment: day,
      type: _constants.Types.FERIA,
      name: Utils.localize({
        key: 'epiphany.after',
        day: day.format('dddd')
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.CHRISTMASTIDE,
          value: Utils.localize({
            key: 'christmastide.season'
          })
        },
        meta: {
          liturgicalColor: '',
          titles: []
        }
      }
    });
  });

  return days;
}; // y: The year (integer)


var _holyWeek = function _holyWeek(y) {
  var dates = Dates.holyWeek(y);
  var days = [];

  _lodash["default"].each(dates, function (date, i) {
    days.push({
      moment: date,
      type: _constants.Types.HOLY_WEEK,
      name: Utils.localize({
        key: 'holyWeek.feria',
        day: date.format('dddd')
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.HOLY_WEEK,
          value: Utils.localize({
            key: 'holyWeek.season'
          })
        },
        meta: {
          liturgicalColor: _constants.LiturgicalColors.PURPLE,
          titles: []
        }
      }
    });
  });

  return days;
}; //================================================================================================
// y: The year (integer)


var advent = function advent(y) {
  var days = [];

  _lodash["default"].each(Dates.daysOfAdvent(y), function (value, i) {
    days.push({
      moment: value,
      type: Utils.getTypeByDayOfWeek(value.day()),
      name: Utils.localize({
        key: _lodash["default"].eq(value.day(), 0) ? 'advent.sunday' : 'advent.feria',
        day: value.format('dddd'),
        week: Math.floor(i / 7) + 1
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.ADVENT,
          value: Utils.localize({
            key: 'advent.season'
          })
        },
        meta: {
          // The proper color of the Third Sunday of Advent is rose. Purple may also be used on these Sundays.
          liturgicalColor: _lodash["default"].eq(Math.floor(i / 7), 2) && _lodash["default"].eq(value.day(), 0) ? _constants.LiturgicalColors.ROSE : _constants.LiturgicalColors.PURPLE,
          titles: []
        }
      }
    });
  }); // Sort dates according to moment


  days = _lodash["default"].sortBy(days, function (item) {
    return item.moment.valueOf();
  }); //=====================================================================
  // PSALTER WEEKS & LITURGICAL COLORS - ADVENT
  //---------------------------------------------------------------------
  // The First Sunday of Advent always begins Week 1 of the Psalter,
  // regardless of which week was previously observed (because the First
  // Sunday of Advent is the first day of the liturgical year, the 4-Week
  // cycle of the Psalter is “reset” to Week 1 on it).
  // For example, as it works out on the liturgical calendar, the last
  // week of Ordinary Time (i.e., the Solemnity of Christ the King, which
  // is the last Sunday of Ordinary Time, and the following Monday through
  // Saturday afternoon) uses Week 2 of the Psalter. Nevertheless, when
  // the First Sunday of Advent arrives, one does not start using Week 3,
  // but rather Week 1 of the Psalter.
  //
  // The proper color of the Advent is purple.
  //=====================================================================
  // Initialize the psalter week counter

  var psalterWeek = 0;

  _lodash["default"].map(days, function (v, k) {
    v.key = _lodash["default"].camelCase(v.name);
    v.data.meta = v.data.meta || {};

    if (_lodash["default"].eq(k % 7, 0)) {
      psalterWeek++;

      if (_lodash["default"].gt(psalterWeek, 4)) {
        psalterWeek = 1;
      }
    } // Psalter week


    v.data.meta.psalterWeek = {
      key: psalterWeek,
      value: _constants.PsalterWeeks[psalterWeek]
    }; // Set default season color if there is no color already set

    v.data.meta.liturgicalColor = v.data.meta.liturgicalColor || _constants.LiturgicalColors.PURPLE;
    return v;
  });

  return _metadata(days);
}; // y: Takes the year (integer)
// christmastideEnds: t|o|e [The mode to calculate the end of Christmastide]
// epiphanyOnJan6: true|false [If true, Epiphany will be fixed to Jan 6] (defaults to false)
// christmastideIncludesTheSeasonOfEpiphany: true|false [If false, excludes the season of epiphany from being included in the season of Christmas]


exports.advent = advent;

var christmastide = function christmastide(y, christmastideEnds) {
  var epiphanyOnJan6 = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : false;
  var christmastideIncludesTheSeasonOfEpiphany = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : true;
  var days = Dates.christmastide(y, christmastideEnds, epiphanyOnJan6);
  var octave = Dates.octaveOfChristmas(y);

  var epiphany = _epiphany(y + 1, epiphanyOnJan6);

  var d = [];
  var count = 0;

  _lodash["default"].each(days, function (day) {
    var dayOfWeek = day.day();

    if (_lodash["default"].eq(dayOfWeek, 0)) {
      count++;
    }

    d.push({
      moment: day,
      type: Utils.getTypeByDayOfWeek(dayOfWeek),
      name: Utils.localize({
        key: _lodash["default"].eq(dayOfWeek, 0) ? 'christmastide.sunday' : 'christmastide.day',
        day: day.format('dddd'),
        count: count
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.CHRISTMASTIDE,
          value: Utils.localize({
            key: 'christmastide.season'
          })
        }
      }
    });
  });

  var o = [];

  _lodash["default"].each(octave, function (day, idx) {
    o.push({
      moment: day,
      type: Utils.getTypeByDayOfWeek(day.day()),
      name: Utils.localize({
        key: 'christmastide.octave',
        count: idx + 1
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.CHRISTMASTIDE,
          value: Utils.localize({
            key: 'christmastide.season'
          })
        }
      }
    });
  }); //==============================================================================
  // Override in order: solemnities, feasts, epiphany and octave of christmas
  // to days of christmas
  //==============================================================================
  // only merge the season of epiphany if the flag is true


  if (christmastideIncludesTheSeasonOfEpiphany === true) {
    d = _lodash["default"].uniqBy(_lodash["default"].union(epiphany, o, d), function (item) {
      return item.moment.valueOf();
    });
  } else {
    d = _lodash["default"].uniqBy(_lodash["default"].union(o, d), function (item) {
      return item.moment.valueOf();
    });
  } // Sort dates according to moment


  d = _lodash["default"].sortBy(d, function (item) {
    return item.moment.valueOf();
  }); //=====================================================================
  // PSALTER WEEKS & LITURGICAL COLORS - CHRISTMAS SEASON
  //---------------------------------------------------------------------
  // If Christmas is on a feria (Monday - Saturday), then the first
  // week of Christmastide will follow the Psalter week of the 4th week
  // of Advent (which is always Psalter Week 4)
  // If Christmas is on a Sunday, the Psalter week will be Week 1
  //
  // The proper color of Christmas is white
  //=====================================================================

  var psalterWeek = 4;

  if (_lodash["default"].eq(_lodash["default"].head(days).day(), 0)) {
    psalterWeek = 0;
  }

  _lodash["default"].map(d, function (v) {
    v.key = _lodash["default"].camelCase(v.name);
    v.data.meta = v.data.meta || {
      titles: []
    };

    if (_lodash["default"].eq(v.moment.day() % 7, 0)) {
      psalterWeek++;

      if (_lodash["default"].gt(psalterWeek, 4)) {
        psalterWeek = 1;
      }
    }

    v.data.meta.psalterWeek = {
      key: psalterWeek,
      value: _constants.PsalterWeeks[psalterWeek]
    }; // Set default season color if there is no color already set

    v.data.meta.liturgicalColor = v.data.meta.liturgicalColor || _constants.LiturgicalColors.WHITE;
    return v;
  });

  return _metadata(d);
}; // y: Takes the year (integer)
// christmastideEnds: t|o|e [The mode to calculate the end of Christmastide]
// epiphanyOnJan6: true|false [If true, Epiphany will be fixed to Jan 6] (defaults to false)


exports.christmastide = christmastide;

var earlyOrdinaryTime = function earlyOrdinaryTime(y, christmastideEnds) {
  var epiphanyOnJan6 = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : false;
  var days = [];

  _lodash["default"].each(Dates.daysOfEarlyOrdinaryTime(y, christmastideEnds, epiphanyOnJan6), function (value, i) {
    days.push({
      moment: value,
      type: _lodash["default"].eq(value.day(), 0) ? _constants.Types.SUNDAY : _constants.Types.FERIA,
      name: Utils.localize({
        key: _lodash["default"].eq(value.day(), 0) ? 'ordinaryTime.sunday' : 'ordinaryTime.feria',
        day: value.format('dddd'),
        week: _lodash["default"].eq(value.day(), 0) ? Math.floor(i / 7) + 2 : Math.floor(i / 7) + 1
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.EARLY_ORDINARY_TIME,
          value: Utils.localize({
            key: 'ordinaryTime.season'
          })
        }
      }
    });
  }); // Sort dates according to moment


  days = _lodash["default"].sortBy(days, function (v) {
    return v.moment.valueOf();
  }); //=====================================================================
  // PSALTER WEEKS & LITURGICAL COLORS - EARLY ORDINARY TIME
  //---------------------------------------------------------------------
  // The first week of Ordinary Time begins with the Monday following
  // the Feast of the Baptism of the Lord (which is the last Sunday of
  // the Christmas Season). Consequently, one starts using Week-1 of the
  // Psalter for that week. The Sunday that follows the first week of
  // Ordinary Time is the Second Sunday of Ordinary Time, so, technically
  // speaking, there is no Sunday that is called the First Sunday of
  // Ordinary Time. This makes sense if you consider the fact that
  // Ordinary Time begins on a Monday, and by the time Sunday rolls
  // around the first week of Ordinary Time has already been completed.
  // In light of all this, although the Feast of the Baptism of the Lord
  // is the last Sunday of the Christmas Season, simply for the purposes
  // of counting Sundays it is counted as the first Sunday of Ordinary Time.
  //
  // The proper color of ordinary time is green
  //=====================================================================

  var psalterWeek = 1;

  _lodash["default"].map(days, function (v) {
    v.key = _lodash["default"].camelCase(v.name);
    v.data.meta = v.data.meta || {
      titles: []
    };

    if (_lodash["default"].eq(v.moment.day() % 7, 0)) {
      psalterWeek++;

      if (_lodash["default"].gt(psalterWeek, 4)) {
        psalterWeek = 1;
      }
    }

    v.data.meta.psalterWeek = {
      key: psalterWeek,
      value: _constants.PsalterWeeks[psalterWeek]
    }; // Set default season color if there is no color alreayd set

    v.data.meta.liturgicalColor = v.data.meta.liturgicalColor || _constants.LiturgicalColors.GREEN;
    return v;
  });

  return _metadata(days);
}; // y: year


exports.earlyOrdinaryTime = earlyOrdinaryTime;

var laterOrdinaryTime = function laterOrdinaryTime(y) {
  // Keep track of the first week in later ordinary time
  // for later use
  var firstWeekOfLaterOrdinaryTime = 0;
  var days = [];

  _lodash["default"].each(Dates.daysOfLaterOrdinaryTime(y).reverse(), function (value, i) {
    // Calculate the week of ordinary time
    // from the last sunday of the year (34th)
    var week = 34 - Math.floor(i / 7);
    firstWeekOfLaterOrdinaryTime = week;
    days.push({
      moment: value,
      type: _lodash["default"].eq(value.day(), 0) ? _constants.Types.SUNDAY : _constants.Types.FERIA,
      name: Utils.localize({
        key: _lodash["default"].eq(value.day(), 0) ? 'ordinaryTime.sunday' : 'ordinaryTime.feria',
        day: value.format('dddd'),
        week: week
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.LATER_ORDINARY_TIME,
          value: Utils.localize({
            key: 'ordinaryTime.season'
          })
        }
      }
    });
  }); // Sort dates according to moment


  days = _lodash["default"].sortBy(days, function (v) {
    return v.moment.valueOf();
  }); //=====================================================================
  // PSALTER WEEKS & LITURGICAL- LATER ORDINARY TIME
  //---------------------------------------------------------------------
  // The second phase of Ordinary Time begins on the Monday that immediately
  // follows Pentecost Sunday (i.e., the last Sunday of the Season of Easter).
  // Note that, because the date of Easter changes from year to year, the
  // specific range of dates for the two phases in Ordinary Time likewise
  // change from year to year. In any event, the Monday of Ordinary Time
  // which immediately follows Pentecost Sunday does not “reset” to Week
  // 1 of the Psalter. Rather, it will simply be the week of the Psalter
  // corresponding to the Sunday to which the Monday belongs.
  //
  // The proper color of Ordinary Time is green.
  //=====================================================================

  var psalterWeek = firstWeekOfLaterOrdinaryTime % 4;

  if (_lodash["default"].eq(psalterWeek, 0)) {
    psalterWeek = 4;
  }

  _lodash["default"].map(days, function (v) {
    v.key = _lodash["default"].camelCase(v.name);
    v.data.meta = v.data.meta || {
      titles: []
    };

    if (_lodash["default"].eq(v.moment.day() % 7, 0)) {
      psalterWeek++;

      if (_lodash["default"].gt(psalterWeek, 4)) {
        psalterWeek = 1;
      }
    }

    v.data.meta.psalterWeek = {
      key: psalterWeek,
      value: _constants.PsalterWeeks[psalterWeek]
    }; // Set default season color if there is no color already set

    if (_lodash["default"].isUndefined(v.data.meta.liturgicalColor)) {
      v.data.meta.liturgicalColor = _constants.LiturgicalColors.GREEN;
    }

    return v;
  });

  return _metadata(days);
}; // y: Takes the year (integer)


exports.laterOrdinaryTime = laterOrdinaryTime;

var lent = function lent(y) {
  var daysOfLent = Dates.daysOfLent(y);
  var sundaysOfLent = Dates.sundaysOfLent(y);
  var days = [];
  var sundays = [];

  _lodash["default"].each(daysOfLent, function (value, i) {
    days.push({
      moment: value,
      type: _constants.Types.FERIA,
      name: Utils.localize({
        key: _lodash["default"].gt(i, 0) && _lodash["default"].lt(i, 4) ? 'lent.day_after_ash_wed' : 'lent.feria',
        day: value.format('dddd'),
        week: Math.floor((i - 4) / 7) + 1
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.LENT,
          value: Utils.localize({
            key: 'lent.season'
          })
        }
      }
    });
  });

  _lodash["default"].each(sundaysOfLent, function (value, i) {
    sundays.push({
      moment: value,
      type: _constants.Types.SUNDAY,
      name: Utils.localize({
        key: 'lent.sunday',
        day: value.format('dddd'),
        week: i + 1
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.LENT,
          value: Utils.localize({
            key: 'lent.season'
          })
        },
        meta: {
          // The proper color of the the Fourth Sunday of Lent is rose. Purple may also be used on these Sundays.
          liturgicalColor: _lodash["default"].eq(i, 3) ? _constants.LiturgicalColors.ROSE : _constants.LiturgicalColors.PURPLE
        }
      }
    });
  });

  var holyWeek = _holyWeek(y); // Override in order: Solemnities, Holy Week and Sundays of Lent to days of Lent


  days = _lodash["default"].uniqBy(_lodash["default"].union(holyWeek, sundays, days), function (v) {
    return v.moment.valueOf();
  }); // Sort dates according to moment

  days = _lodash["default"].sortBy(days, function (v) {
    return v.moment.valueOf();
  }); //=====================================================================
  // PSALTER WEEKS & LITURGICAL COLORS - LENT & HOLY WEEK
  //---------------------------------------------------------------------
  // Lent begins on Ash Wednesday and therefore in the middle of a week
  // rather than on a Sunday. Regardless of which week of the Psalter is
  // being observed on the Sunday through Tuesday immediately preceding
  // Ash Wednesday, Ash Wednesday (and the Thursday through Saturday
  // afternoon following it) are designated as Week 4 of the Psalter.
  //
  // The proper color of the Lent is purple.
  //=====================================================================

  var psalterWeek = 4;

  _lodash["default"].map(days, function (v) {
    v.key = _lodash["default"].camelCase(v.name);
    v.data.meta = v.data.meta || {
      titles: []
    };

    if (_lodash["default"].eq(v.moment.day() % 7, 0)) {
      psalterWeek++;

      if (_lodash["default"].gt(psalterWeek, 4)) {
        psalterWeek = 1;
      }
    }

    v.data.meta.psalterWeek = {
      key: psalterWeek,
      value: _constants.PsalterWeeks[psalterWeek]
    }; // Set default season color if there is no color already set

    v.data.meta.liturgicalColor = v.data.meta.liturgicalColor || _constants.LiturgicalColors.PURPLE;
    return v;
  });

  days = _metadata(days); // _.each( days, function( v ) {
  //   console.log(
  //     v.moment.format('ddd, DD MMM YY'),
  //     '|', _.padRight( v.data.meta.liturgicalColor.key, 6 ),
  //     '|', _.padRight( v.data.season.value, 9 ),
  //     '|', _.padRight( v.data.meta.psalterWeek.value, 8 ),
  //     '|', _.padRight( v.type, 13 ),
  //     '|', _.padRight( v.data.calendar.week, 2 ),
  //     '|', _.padRight( v.data.calendar.day, 3 ),
  //     '|', v.name
  //   );
  // });

  return days;
}; // Takes the last 3 days of holy week which form the 3 days of Easter Triduum
// y: Takes the year (integer)


exports.lent = lent;

var easterTriduum = function easterTriduum(y) {
  return _lodash["default"].takeRight(_holyWeek(y), 3);
}; // Takes the days between Easter Sunday and Divine mercy sunday (inclusive) to form the easter octave
// y: Takes the year (integer)


exports.easterTriduum = easterTriduum;

var easterOctave = function easterOctave(y) {
  return _lodash["default"].take(eastertide(y), 8);
}; // y: Takes the year (integer)


exports.easterOctave = easterOctave;

var eastertide = function eastertide(y) {
  var d = Dates.daysOfEaster(y);
  var s = Dates.sundaysOfEaster(y);
  var days = [];

  _lodash["default"].each(d, function (value, i) {
    days.push({
      moment: value,
      type: _lodash["default"].gt(i, 0) && _lodash["default"].lt(i, 7) ? _constants.Types.SOLEMNITY : _constants.Types.FERIA,
      name: Utils.localize({
        key: _lodash["default"].gt(i, 0) && _lodash["default"].lt(i, 7) ? 'eastertide.octave' : 'eastertide.feria',
        day: value.format('dddd'),
        week: Math.floor(i / 7) + 1
      }),
      data: {
        prioritized: true,
        season: {
          key: _constants.LiturgicalSeasons.EASTER,
          value: Utils.localize({
            key: 'eastertide.season'
          })
        }
      }
    });
  });

  var sundays = [];

  _lodash["default"].each(s, function (value, i) {
    sundays.push({
      moment: value,
      type: _constants.Types.SUNDAY,
      name: Utils.localize({
        key: 'eastertide.sunday',
        day: value.format('dddd'),
        week: i + 1
      }),
      data: {
        season: {
          key: _constants.LiturgicalSeasons.EASTER,
          value: Utils.localize({
            key: 'eastertide.season'
          })
        }
      }
    });
  }); // Insert Solemnities and Sundays of Lent to days of Easter


  days = _lodash["default"].uniqBy(_lodash["default"].union(sundays, days), function (v) {
    return v.moment.valueOf();
  }); // Sort dates according to moment

  days = _lodash["default"].sortBy(days, function (v) {
    return v.moment.valueOf();
  }); //=====================================================================
  // PSALTER WEEKS & LITURGICAL COLOR - EASTER
  //---------------------------------------------------------------------
  // The first eight days of the Easter Season make up the octave of Easter
  // and are celebrated as Solemnities of the Lord.
  // During this time, Easter Octave readings are used in favour of the
  // standard psalter readings.
  // The psalter is resumed at week 2 on the Monday following Divine
  // Mercy Sunday
  //
  // The proper color of Easter is white
  //=====================================================================

  var psalterWeek = 2;

  _lodash["default"].map(days, function (v, k) {
    v.key = _lodash["default"].camelCase(v.name);
    v.data.meta = {
      titles: []
    };

    if (_lodash["default"].lt(k, 8)) {
      v.data.meta.psalterWeek = {
        key: 5,
        value: _constants.PsalterWeeks[5]
      };
    } else {
      if (_lodash["default"].eq(v.moment.day() % 7, 0)) {
        psalterWeek++;

        if (_lodash["default"].gt(psalterWeek, 4)) {
          psalterWeek = 1;
        }
      }

      v.data.meta.psalterWeek = {
        key: psalterWeek,
        value: _constants.PsalterWeeks[psalterWeek]
      };
    } // Set default season color if there is no color already set


    v.data.meta.liturgicalColor = v.data.meta.liturgicalColor || _constants.LiturgicalColors.WHITE;
    return v;
  });

  days = _metadata(days); // _.each( days, function( v ) {
  //   console.log(
  //     v.moment.format('ddd, DD MMM YY'),
  //     '|', _.padRight( v.data.meta.liturgicalColor.key, 6 ),
  //     '|', _.padRight( v.data.season.value, 9 ),
  //     '|', _.padRight( v.data.meta.psalterWeek.value, 8 ),
  //     '|', _.padRight( v.type, 13 ),
  //     '|', v.name
  //   );
  // });

  return days;
};

exports.eastertide = eastertide;
