// romcal/dist/index.js
"use strict";

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Calendar", {
  enumerable: true,
  get: function get() {
    return _lib.Calendar;
  }
});
Object.defineProperty(exports, "Celebrations", {
  enumerable: true,
  get: function get() {
    return _lib.Celebrations;
  }
});
Object.defineProperty(exports, "Dates", {
  enumerable: true,
  get: function get() {
    return _lib.Dates;
  }
});
Object.defineProperty(exports, "Seasons", {
  enumerable: true,
  get: function get() {
    return _lib.Seasons;
  }
});
Object.defineProperty(exports, "Utils", {
  enumerable: true,
  get: function get() {
    return _lib.Utils;
  }
});
Object.defineProperty(exports, "Cycles", {
  enumerable: true,
  get: function get() {
    return _constants.Cycles;
  }
});
Object.defineProperty(exports, "LiturgicalColors", {
  enumerable: true,
  get: function get() {
    return _constants.LiturgicalColors;
  }
});
Object.defineProperty(exports, "PsalterWeeks", {
  enumerable: true,
  get: function get() {
    return _constants.PsalterWeeks;
  }
});
Object.defineProperty(exports, "LiturgicalSeasons", {
  enumerable: true,
  get: function get() {
    return _constants.LiturgicalSeasons;
  }
});
Object.defineProperty(exports, "Titles", {
  enumerable: true,
  get: function get() {
    return _constants.Titles;
  }
});
Object.defineProperty(exports, "Types", {
  enumerable: true,
  get: function get() {
    return _constants.Types;
  }
});
exports.Locales = exports["default"] = exports.queryFor = exports.calendarFor = exports.Localizations = exports.Countries = void 0;

var _lodash = _interopRequireDefault(require("lodash"));

var _moment = _interopRequireDefault(require("moment"));

var _momentRange = require("moment-range");

var _lib = require("./lib");

var _constants = require("./constants");

var Locales = _interopRequireWildcard(require("./locales"));

exports.Locales = Locales;

var Calendars = _interopRequireWildcard(require("./calendars"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/*
    The MIT License (MIT)

    Copyright (c) 2014 Pereira, Julian Matthew

    Permission is hereby granted, free of charge, to any person obtaining a copy
    of this software and associated documentation files (the "Software"), to deal
    in the Software without restriction, including without limitation the rights
    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
    copies of the Software, and to permit persons to whom the Software is
    furnished to do so, subject to the following conditions:

    The above copyright notice and this permission notice shall be included in
    all copies or substantial portions of the Software.

    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
    THE SOFTWARE.
*/
var moment = (0, _momentRange.extendMoment)(_moment["default"]);

var Localizations = _lodash["default"].keys(_lodash["default"].mapKeys(Locales, function (v, k) {
  return _lodash["default"].kebabCase(k);
}));

exports.Localizations = Localizations;

var Countries = _lodash["default"].keys(Calendars); // Export an array of countries for external use
// Export an array of locales for external use


exports.Countries = Countries;
var calendarFor = _lib.Calendar.calendarFor,
    queryFor = _lib.Calendar.queryFor;
exports.queryFor = queryFor;
exports.calendarFor = calendarFor;
calendarFor();
// Default entry point is exported as Romcal
var _default = _lib.Calendar;
exports["default"] = _default;
